### PR TITLE
ISSUE [#5709] - Removed duplicate message during login authorization

### PR DESCRIPTION
**Pull request by twhitloc**
Attempted to fix duplicate authentication warnings during login

---

-Go to sitename/administrator
-Enter valid username & enter invalid password
-Click on Login button twice, thrice as much as you can ;)
-**Application will display multiple messages!**

Solution:
-Button should trigger once.

screen shot 2015-01-13 at 08 37 30

### CODE REVIEW

**Comment by twhitloc** [comment]
I do not understand these white space errors that are preventing the Travis build.

File: libraries/cms/application/cms.php [hunk 1]
```diff
@@ -823,24 +822,28 @@ public function login($credentials, $options = array())
 					{
 						return false;
 					}
+					// If the user has duplicated request for login ignore the duplicate attempt's message.
+					if($previousauth != $authorisation)
```
Comment (wilsonge) [inline libraries/cms/application/cms.php]: You need a space between the if and the open bracket here for travis :)

**Comment by twhitloc** [comment]
Thank you for the help!  I was trying to figure it out but the comment that showed up was a bit confusing to me :)

File: libraries/cms/application/cms.php [hunk 2]
```diff
@@ -823,24 +822,28 @@ public function login($credentials, $options = array())
...
+								break;
 
-						default:
-							return JError::raiseWarning('102004', JText::_('JLIB_LOGIN_AUTHORISATION'));
+							default:
+								return JError::raiseWarning('102004', JText::_('JLIB_LOGIN_AUTHORISATION'));
 
-							break;
+								break;
+						}
```
Comment (wilsonge) [inline libraries/cms/application/cms.php]: There's an extra tab at the end of this line you need to remove

File: libraries/cms/application/cms.php [hunk 3]
```diff
@@ -823,24 +822,28 @@ public function login($credentials, $options = array())
 					{
 						return false;
 					}
+					// If the user has duplicated request for login ignore the duplicate attempt's message.
+					if($previousauth != $authorisation)
+					{
```
Comment (wilsonge) [inline libraries/cms/application/cms.php]: There's an extra tab at the end of this line you need to remove

**Comment by wilsonge** [comment]
It's ok :) I remember how confusing it was to me when I started. If you fix these 3 issues I've commented travis should be happy!

**Comment by twhitloc** [comment]
Thank you, I will fix these issues and upload it later!

**Comment by infograf768** [comment]
Please add blank line before each `foreach` or `if` or before any comment only before these.

**Comment by twhitloc** [comment]
Thank you infograf, I have added the spacing before the comment.  Sorry for the delay.

**Comment by bertmert** [comment]
I cannot reproduce this issue with current staging. Made lots of different tests and always got just 1 message. Is this error browser dependent?
But @brianteeman confirmed it https://github.com/joomla/joomla-cms/issues/5709#issuecomment-69811947

**Comment by brianteeman** [comment]
Just double checked 
I dont see it in firefox at all BUT in chrome and safari I can replicate it for the blank password message but not for the username/password do not match message

**Comment by jduerscheid** [comment]
Issue successful tested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by designbengel** [comment]
No, sorry but i still have the multiple messages after applying the patch...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by jduerscheid** [comment]
@designbengel: Which OS and Browser did you used for the test? I used Windows8, Firefox and Chrome:<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by designbengel** [comment]
I tested on Safari 8.0.5 / Mac Os X Yosemite 10.10.3

**Comment by larapetersen** [comment]
Tested on Chrome, i still have the multiple messages too after applying the patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by RickR2H** [comment]
Bug confirmed but the patch doesn't work.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by RickR2H** [comment]
But... is this a bug? When I click 3 times it means that I have 3 faulty attempts so 3 errors.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by roland-d** [comment]
I also tested this and the PR doesn't solve the issue for me. Seeing the notice 3 times isn't an issue for me, if you click 3x you can get the message 3x. However what is an issue is that if you click 3x with a valid username/password you can get this message:
> The most recent request was denied because it contained an invalid security token. Please refresh the page and try again.

As you said the button should be clickable once, this is better achieved using JavaScript like we do with other buttons, for example the article editing. Could you update the PR to give that a try?

**Comment by chrisdavenport** [comment]
This issue should be solved in JavaScript.  Thanks for raising this issue and for your contribution, but in the absence of a suitable PR for that, I'm closing this issue at this time.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>.</sub>

**Comment by zero-24** [comment]
Set to "closed" on behalf of @zero-24 by The <a href="https://github.com/joomla/jissues">JTracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/6832">issues.joomla.org/joomla-cms/6832</a>

**Pull request closed without merging**